go/master/client.go

```diff
@@ -20,6 +25,51 @@ type Client struct {
 	ch   chan []byte
 }
 
+// MasterAddresser provide master address
+type masterAddresser struct {
+	client    *clientv3.Client
+	timeout   time.Duration
+	endpoints []string
+}
+
+// Address return the address
+func (m masterAddresser) Address() string {
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+		resp, err := m.client.Get(ctx, masterAddrPath)
```

Comment:
I originally thought client can poll etcd to get the latest master address. But I think @typhoonzero 's idea is better.
Because `Addresser` is an polling interface, to get notification about the change rather than polling to get the change. We need to change the interface. One proposal would be:
Change:
```go
func NewClient(addr Addresser, bufSize int)
```
To:
```go
func NewClient(addrCh <-chan string, bufSize int)
```

And maybe change moniterMaster to:
```go
func (c *Client) monitorMaster(addrCh <-chan string) {
        lastMaster := ""
        for curMaster := range addrCh {
                // connect to the new address once address changed.                                                                                                                                                                         
                if curMaster != lastMaster {
                        if curMaster == "" {
                                err := c.conn.Close()
                                if err != nil {
                                        log.Errorln(err)
                                }
                        } else {
                                err := c.conn.Connect(curMaster)
                                if err != nil {
                                        log.Errorln(err)

                                        // connect to addr failed, set                                                                                                                                                                      
                                        // to last known addr in order                                                                                                                                                                      
                                        // to retry next time.                                                                                                                                                                              
                                        curMaster = lastMaster
                                }

                        }
                }
	        lastMaster = curMaster
        }
}
```